### PR TITLE
Allow TokenManager keeping several token pairs

### PR DESCRIPTION
**Pull request by KutsenkoA**
1. Remove the whole auth module: 
  - UMS module will be in charge of user auth
  - TokenManager are able to auth with apikey
2. Refactoring TokenManager and Storage components

### CODE REVIEW

File: src/api/core/tokenManager.ts [hunk 1]
```diff
@@ -68,6 +71,14 @@ export const AuthOptions = new InjectionToken<IAuthOptions>("IAuthOptions");
...
  */
 export interface IAuthToken {
+  /**
+   * Alias for the token pair
+   */
+  auth: string;
+  /**
+   * Is it default token pair?
+   */
+  default: boolean;
```
Comment (marlosin) [inline src/api/core/tokenManager.ts]: I'd rather name it as `isDefault`

Comment (KutsenkoA) [reply src/api/core/tokenManager.ts]: ha-ha, I suddenly realised that I don't need this interface anymore 👍

File: src/api/core/componentStorage.ts [hunk 2]
```diff
@@ -1,53 +1,90 @@
...
-  getTokens(): Promise<IAuthToken>;
-  clear(): Promise<void>;
+
+  /**
+   * Save token pair into the storage
+   * @param auth {string} Alias for the token pair
+   * @param tokens {Tokens} Token pair to save
+   * @param defaults {boolean} Use this token pair by default if true
+   */
+  setTokens(auth: string, tokens: Tokens, defaults: boolean): void;
```
[the remark this answers is not in the document]
Comment (KutsenkoA) [reply src/api/core/componentStorage.ts]: 🤦‍♂️

File: src/api/core/componentStorage.spec.ts [hunk 3]
```diff
@@ -1,10 +1,11 @@
 // tslint:disable:no-string-literal
+// @ts-ignore
```
Comment (marlosin) [inline src/api/core/componentStorage.spec.ts]: can you test adding `@types/faker` package to check if ts compiler would still complain?

File: src/api/core/tokenManager.ts (hunk 1 above)
Comment (marlosin) [reply src/api/core/tokenManager.ts]: that's even better :smile:

**Pull request merged**